### DISCUSSION TITLE
Force popup to landscape mode?

### DISCUSSION THREAD

**Discussion by ahgraber** [uBlockOrigin/uBlock-issues#2419] [General]
Hello!  Is there any way to force the popup to landscape mode?  There are times when ublock seems to not realize that there is horizontal room, and loads itself vertically.  This is generally not a _problem_, other than I strongly prefer the landscape view for less scrolling.  It happens intermittently -- rarely in FF, sometimes in ungoogled-chromium, safari, and most recently in kagi (which I realize is probably an unsupported browser)

<img width="408" alt="Screen Shot 2022-12-21 at 21 12 10" src="https://user-images.githubusercontent.com/24922003/209040130-ef216156-907c-4869-9a4c-2fa359146eb9.png">

Apologies if I've missed this in my search for this issue/solutions!

**Comment by partingscientist** [comment]
Correct me if I'm wrong, but this seems similar to #2385 that should be fixed in development build. Perhaps the maintainers might want to confirm this.

**Comment by uBlock-user** [comment]
@ahgraber Try the dev build

**Reply by ahgraber** [reply]
Thanks for the suggestion.  I installed 1.45.3rc6 but the issue remains.  
I've also [submitted this issue for review on Kagi Orion](https://orionfeedback.org/d/3977-ublock-origin-popup-pane-does-not-render-correctly), as this is a constant issue in the Kagi Orion browser and intermittent/rare in FF/Chromium.

**Reply by gorhill** [reply]
First would be to find out if Kagi Orion imposes a hard limit on the width for popup panels of extensions, in which case this can't be solved here. Looking at uBO's CSS for the popup panel, the minimum width for the normal popup panel is 18em + 30em = 48em. If the viewport ends up being below this value, uBO will toggle automatically in "portrait" mode, which is what you get.

---

Maybe see if changing the advanced setting [`popupFontSize`](https://github.com/gorhill/uBlock/wiki/Advanced-settings#popupfontsize) to something smaller than `14px` makes a difference.

**Reply by ahgraber** [reply]
I tried popupFontSize down to 8px with no success.

Thanks for the pointers re: popup viewport width!  I'll pass it along

**Reply by gorhill** [reply]
I could add an advanced setting to force the popup panel in landscape mode, but ultimately it would be nice to understand the root of the issue.

**Reply by imyelmo** [reply]
How can you  force the popup panel in landscape mode? I'm facing this problem right now, I would like to find a provisional solution for this issue. Thanks

**Reply by gorhill** [reply]
Be sure your are not using a zoom factor which causes insufficient horizontal space for the popup panel to stay in landscape mode. The popup panel automatically fall back to portrait mode when it detects that part of its DOM falls outside the viewport on the left. What's your display specs and zoom factor in you browser?

**Reply by imyelmo** [reply]
Display 2560x1440 pixels
Zoom 100%

I don't know if you need further information

**Reply by gorhill** [reply]
Can you provide a screenshot, troubleshooting information and which exact browser/version you are using?

**Comment by imyelmo** [comment]
MacOS Sonoma 14.4 
Orion Browser Version 0.99.126.4.1.10-rc (WebKit 619.1.1)
Firefox Extension uBlock Origin 1.56.0

![Screenshot 2024-03-11 at 13 52 15](https://github.com/uBlockOrigin/uBlock-issues/assets/11572612/777102d3-b094-4045-896a-ce3db36b6fc3)

**Reply by gorhill** [reply]
Orion Browser is not officially supported, it's not something I can test. My guess is that the browser does not provide enough horizontal space for uBO's popup panel, which forces it to render in portrait mode.

**Reply by ghost** [reply]
Maybe lower value in https://github.com/uBlockOrigin/uBlock-issues/wiki/Advanced-settings#popupfontsize will help

**Reply by imyelmo** [reply]
A lower value does not solve the problem (checked). I understand that the browser is not officially supported. Actually, that is the reason because of I asked for a temporal solution to force landscape mode.

**Reply by gorhill** [reply]
https://github.com/gorhill/uBlock/commit/0d77ccded7

**Reply by imyelmo** [reply]
Thanks, I really appreciate it. Probably, more users will be happy in case of issues with independence of the platform. Waiting for the next release!

**Reply by gorhill** [reply]
You can test dev build b15 to confirm this helps: <https://github.com/gorhill/uBlock/releases/download/1.56.1b15/uBlock0_1.56.1b15.firefox.signed.xpi>

**Reply by imyelmo** [reply]
I tested it and it is not working. If I configure landscape mode, the lefts side of the landscape mode with the information about the different script disappears.

<details><summary>Screenshot</summary>

![Screenshot 2024-03-12 at 11 52 05](https://github.com/uBlockOrigin/uBlock-issues/assets/11572612/a14cfec7-8cd7-4b3c-abb0-1103957b1edb)

</details>

**Reply by gorhill** [reply]
Then I would say it's an Orion browser issue, it should resize its popup panel container according to the size of uBO's document.

**Reply by ghost** [reply]
In Nightly in portrait mode, content is not scrollable.

**Reply by gorhill** [reply]
I just added that setting to try to help with the Orion browser, I will not put more time into this, consider it's experimental, and I might just remove it since it didn't help.

**Reply by imyelmo** [reply]
I think that it is an interesting setting to debugging extensions in web browsers. In Europe, iOS browsers are going to be open to web engines different from webkit. Thus,  I think is an useful setting in the near future.

**Reply by gorhill** [reply]
The popup panel is already working fine on mobile Firefox, the issue re. portrait not scrolling is specific to the new setting I added, and probably occurs because the `mobile` class is not present in the `html` element -- but this will work fine on mobile because that class is added by uBO when it detects a mobile environment.

**Reply by gorhill** [reply]
> Nightly in portrait mode, content is not scrollable

Or did you mean in Firefox for Android Nightly?

**Reply by ghost** [reply]
Desktop. `modifyWebextFlavor +mobile`  does not help.

**Reply by gorhill** [reply]
`modifyWebextFlavor` applies only for the background process, not auxiliary pages -- it was introduced so that filter lists could be compiled with `env_mobile` evaluating to `true`.
